libp2p/protocols/identify.nim

```diff
@@ -42,8 +42,13 @@ type
     agentVersion*: Option[string]
     protos*: seq[string]
 
+  CurrentAddressProvider* =
```

Comment:
This is probably no the correct PR to work on this change, this is a bit more involved that and goes beyond just identify. Let's move it out to it's own branch.

Reply:
Sure, but for now it involves only identity I think. I will move it out but likely will need this branch